### PR TITLE
Use `git ls-files` in a few places instead of `find`

### PR DESCRIPTION
**Pull request by thockin**
/kind cleanup
/sig architecture

```release-note
NONE
```

### CODE REVIEW

File: test/instrumentation/stability-utils.sh [hunk 1]
```diff
@@ -29,37 +29,46 @@ stability_check_setup() {
...
+        ':!:./test/instrumentation' \
+        ':(glob)**/*.go' \
+        "$@"
 }
 
-find_test_files() {
-  find './test/instrumentation' -wholename '**/*.go'
+function find_test_files() {
+  git ls-files -cmo --exclude-standard \
+      ':!:*/*_test.go' \
```
Comment (aojea) [inline test/instrumentation/stability-utils.sh:49]: Why do you exclude test.go files here? It seems that previously were not excluded

**Comment by aojea** [comment]
2 first commit are straightforward, last one is 🤔

File: test/instrumentation/stability-utils.sh (hunk 1 above)
Comment (thockin) [reply test/instrumentation/stability-utils.sh:49]: yeah, that looks like a mistake!

**Review by thockin — COMMENTED** [review]
New push

File: hack/update-gofmt.sh [hunk 2]
```diff
@@ -27,20 +27,16 @@ kube::golang::verify_go_version
...
-      \) -prune \
-    \) -name '*.go'
+function git_find() {
+    # Similar to find but faster and easier to understand.  We want to include
+    # modified and untracked files because this might be running against code
+    # which is not tracked by git yet.
+    git ls-files -cmo --exclude-standard \
+        ':!:vendor/*' \
+        ':!:third_party/*' \
+        ':!:*/testdata/*' \
```
Comment (aojea) [inline hack/update-gofmt.sh:37]: should we exclude testdata

```
 git ls-files -cmo  --exclude-standard | grep testdata | grep \.go$
staging/src/k8s.io/kubectl/pkg/cmd/edit/testdata/record.go
test/conformance/testdata/embed.go
test/instrumentation/testdata/pkg/kubelet/metrics/metrics.go
test/instrumentation/testdata/staging/src/k8s.io/metrics/metrics.go
test/typecheck/testdata/bad/bad.go
test/typecheck/testdata/good/_ignore/bad.go
test/typecheck/testdata/good/good.go
test/typecheck/testdata/good/testdata/bad.go

```

File: hack/update-netparse-cve.sh [hunk 3]
```diff
@@ -45,22 +45,17 @@ popd >/dev/null
...
-      \) -prune \
-    \) -name '*.go'
+function git_find() {
+    # Similar to find but faster and easier to understand.  We want to include
+    # modified and untracked files because this might be running against code
+    # which is not tracked by git yet.
+    git ls-files -cmo --exclude-standard \
+        ':!:vendor/*' \
+        ':!:third_party/*' \
+        ':!:*/testdata/*' \
```
Comment (aojea) [inline hack/update-netparse-cve.sh:55]: ditto

File: test/instrumentation/stability-utils.sh [hunk 4]
```diff
@@ -29,37 +29,45 @@ stability_check_setup() {
...
-  \)
+function find_files_to_check() {
+    # Similar to find but faster and easier to understand.  We want to include
+    # modified and untracked files because this might be running against code
+    # which is not tracked by git yet.
+    git ls-files -cmo --exclude-standard \
+        ':!:vendor/*' \
+        ':!:third_party/*' \
+        ':!:hack/*' \
+        ':!:*/testdata/*' \
```
Comment (aojea) [inline test/instrumentation/stability-utils.sh:40]: ditto

Comment (aojea) [reply test/instrumentation/stability-utils.sh:40]: it seems that testdata files are also referred in this script
```
 git ls-files -cmo  --exclude-standard | grep testdata | grep \.go$
staging/src/k8s.io/kubectl/pkg/cmd/edit/testdata/record.go
test/conformance/testdata/embed.go
test/instrumentation/testdata/pkg/kubelet/metrics/metrics.go
test/instrumentation/testdata/staging/src/k8s.io/metrics/metrics.go
test/typecheck/testdata/bad/bad.go
test/typecheck/testdata/good/_ignore/bad.go
test/typecheck/testdata/good/good.go
test/typecheck/testdata/good/testdata/bad.go

```

File: hack/update-gofmt.sh (hunk 2 above)
Comment (thockin) [reply hack/update-gofmt.sh:37]: Good question - I wanted to retain the same behavior, but testdata is always questionable.  Will revert that part

File: test/instrumentation/stability-utils.sh (hunk 4 above)
Comment (thockin) [reply test/instrumentation/stability-utils.sh:40]: ACK.  I must have cut-pasted this block and forgotten to remove it or something, thanks.

**Comment by thockin** [comment]
New push is up

File: test/instrumentation/stability-utils.sh [hunk 5]
```diff
@@ -29,37 +29,44 @@ stability_check_setup() {
...
-        -o -wholename '*/third_party/*' \
-        -o -wholename '*/vendor/*' \
-        -o -wholename '*/hack/*' \
-        -o -wholename '*_test.go' \
-        -o -wholename './test/instrumentation/*.go' \
-        \) -prune \
-    \) \
-    \( -wholename '**/*.go' \
-  \)
+function find_files_to_check() {
```
Comment (aojea) [inline test/instrumentation/stability-utils.sh:32]: the diff is different
```
 diff filesfind.txt filesgit.txt
2829a2830
> staging/src/k8s.io/apiextensions-apiserver/examples/client-go/hack/tools.go
3226a3228,3230
> staging/src/k8s.io/apimachinery/third_party/forked/golang/json/fields.go
> staging/src/k8s.io/apimachinery/third_party/forked/golang/netutil/addr.go
> staging/src/k8s.io/apimachinery/third_party/forked/golang/reflect/deep_equal.go
5521a5526,5527
> staging/src/k8s.io/client-go/third_party/forked/golang/template/exec.go
> staging/src/k8s.io/client-go/third_party/forked/golang/template/funcs.go
6012a6019
> staging/src/k8s.io/code-generator/third_party/forked/golang/reflect/type.go
6219a6227
> staging/src/k8s.io/kube-aggregator/hack/tools.go
6822a6831
> staging/src/k8s.io/metrics/hack/tools.go
6915a6925
> staging/src/k8s.io/noderesourcetopology-api/hack/tools.go
7004a7015
> staging/src/k8s.io/sample-apiserver/hack/tools.go
7077a7089
> staging/src/k8s.io/sample-controller/hack/tools.go

```

is this ok?

Comment (aojea) [reply test/instrumentation/stability-utils.sh:32]: I see, this new `find` doesn't suppress the `third_party` and `hack` folders of the staging repos

Comment (aojea) [reply test/instrumentation/stability-utils.sh:32]: should we tune the pathspec `Magic signatures` ?

File: hack/update-gofmt.sh [hunk 6]
```diff
@@ -27,20 +27,15 @@ kube::golang::verify_go_version
...
-        -o -wholename './staging/src/k8s.io/client-go/*vendor/*' \
-      \) -prune \
-    \) -name '*.go'
+function git_find() {
+    # Similar to find but faster and easier to understand.  We want to include
+    # modified and untracked files because this might be running against code
+    # which is not tracked by git yet.
+    git ls-files -cmo --exclude-standard \
+        ':!:vendor/*' \
+        ':!:third_party/*' \
```
Comment (aojea) [inline hack/update-gofmt.sh:36]: This doesn't match the `third_party` inside the staging repos

```
git ls-files -cmo --exclude-standard         ':!:vendor/*'         ':!:third_party/*'  | grep third_party
LICENSES/third_party/forked/golang/LICENSE
LICENSES/third_party/forked/gonum/graph/LICENSE
LICENSES/third_party/forked/gotestsum/LICENSE
LICENSES/third_party/forked/gotestsum/NOTICE
LICENSES/third_party/forked/shell2junit/LICENSE
LICENSES/third_party/multiarch/qemu-user-static/LICENSE
LICENSES/vendor/github.com/google/cadvisor/third_party/containerd/LICENSE
```

File: test/instrumentation/stability-utils.sh (hunk 5 above)
Comment (thockin) [reply test/instrumentation/stability-utils.sh:32]: ooh, good catch.

Comment (thockin) [reply test/instrumentation/stability-utils.sh:32]: hmm, I don't know how to spec a glob an negation together.  Have to go AFK for a bit, will investigate

**Comment by thockin** [comment]
New push is up - I didn't figure out how to do a negated glob, so I did it with 2 rules - if I figure it out we can revisit :)

**Pull request merged**